### PR TITLE
Vendor c/image after https://github.com/containers/image/pull/1816

### PR DESCRIPTION
**Pull request by mtrmac**
<s>WIP, this is primarily to expose the code to flakes, and to collect production-environment error values and other data.

Look for `BODYREADER DEBUG` messages in logs, they are intentionally _very_ loud right now</s>

#### Does this PR introduce a user-facing change?

```release-note
Automatically transparently requires aborted pull connections (in some strictly limited situations)
```

### CODE REVIEW

**Comment by mtrmac** [comment]
Cc: @edsantiago

**Comment by mtrmac** [comment]
@edsantiago the test logs (very reasonably) don’t include output from Podman commands ran as part of the tests if the tests succeed. Do you know if there is some pre-existing, reasonably convenient, trick I could use to log things from deep inside Podman code in a way that would then be possible to view on test success? Maybe syslog?

**Comment by mtrmac** [comment]
`dial tcp: lookup cdn03.quay.io: no such host` are quite frequent, and _not_ the target of this PR.

I mean, are we supposed to detect and workaround consistently incorrect DNS in redirect targets? It’s probably _possible_ (after we successfully ping `quay.io/v2/`, any future “DNS not found” errors can be assumed to be infrastructure), but … seriously?

**Comment by edsantiago** [comment]
>some pre-existing, reasonably convenient, trick I could use to log things from deep inside Podman code in a way that would then be possible to view on test success? Maybe syslog?

From a BATS script, you can echo to fd3, prepending a hash:
```bash
    echo "# debug: $output" >&3
```
From podman itself, you can't, because I take pains to close fd3 before running podman. But journal logs are available in CI, via a series of clicks on the Cirrus page (near the bottom of the accordions).

>I mean, are we supposed to detect and workaround consistently incorrect DNS in redirect targets? 

I reported the flake to quay on Thursday. Haven't made it past the "works for me" phase yet.

**Comment by mtrmac** [comment]
One caught failure:
```
BODYREADER DEBUG: Reading blob body, decision inputs: lastRetryOffset 0, offset 60631823, error &errors.errorString{s:\"unexpected EOF\"}
```
Good, so the heuristics did match at least in this one case.

**Comment by mtrmac** [comment]
```
BODYREADER DEBUG: Reading blob body, decision inputs: lastRetryOffset 0, offset 260352, error &errors.errorString{s:"unexpected EOF"}
```
The current heuristic requires 1 MB, and did not trigger here.

**Comment by mtrmac** [comment]
Analyzing test runs from Jan 30 (and making a record here so that we have _some_ sort of statistics):

Quite a few tests are broken by the extra logging; I need to tune that down.

---

```
BODYREADER DEBUG: Reading blob body, decision inputs: lastRetryOffset 0, offset 2771195, time diff 0 ms, error &errors.errorString{s:\"unexpected EOF\"}
```
so “immediately” after successfully receiving data (possibly buffered locally? OTOH clearly not a stalled connection), we get unexpected EOF.

On the positive side, reconnecting seems to have worked (the first? recorded instance).

---

```
BODYREADER DEBUG: Reading blob body, decision inputs: lastRetryOffset 0, offset 7403776, time diff 0 ms, error &errors.errorString{s:\"unexpected EOF\"}"
```

also seems to have worked.

**Comment by mtrmac** [comment]
```
vendor/github.com/openshift/imagebuilder/dockerfile/parser/parser.go:122:12: undefined: system.LCOWSupported
```
https://github.com/openshift/imagebuilder/issues/238

Note to self: a backport to the 5.24 branch might be in order…

**Review by vrothberg — COMMENTED** [review]
LGTM

I am also in favor of a backport.  If we backport, today is a good time to get it in before the 4.4.1 release.

**Comment by mtrmac** [comment]
Merging #17342 makes this PR, going back to a c/image 5.24 stable branch, plausible but less unattractive. Something like the closed #17405, rolling c/image even further forward, would make Podman smaller (and on the main branch it’s closer to the long-term direction).

**Comment by mtrmac** [comment]
Due to #17342 being merged, just using a c/image 5.24 branch would be a revert. So, this PR now just rolls c/image forward to include the `main`-branch merged PR. That brings several other dependency updates.

So, this also:
- includes unreleased https://github.com/openshift/imagebuilder/pull/246 to work with the updated docker/docker dependency.
- updates some references to newly deprecated docker/docker symbols.
    
Ready for review and possible merging.

**Comment by edsantiago** [comment]
>podman grew by 91528 bytes; max allowed is 51200

Ouch.

Restarted system-test flake, it's the `dial tcp: lookup cdn03.quay.io: no such host` one.

File: go.mod [hunk 1]
```diff
@@ -14,15 +14,15 @@ require (
...
-	github.com/containers/image/v5 v5.24.1-0.20230202144111-a49c94a010be
+	github.com/containers/image/v5 v5.24.1-0.20230208161124-34839152eb48
 	github.com/containers/ocicrypt v1.1.7
 	github.com/containers/psgo v1.8.0
 	github.com/containers/storage v1.45.3
 	github.com/coreos/go-systemd/v22 v22.5.0
 	github.com/coreos/stream-metadata-go v0.0.0-20210225230131-70edb9eb47b3
 	github.com/cyphar/filepath-securejoin v0.2.3
 	github.com/digitalocean/go-qemu v0.0.0-20210326154740-ac9e0b687001
-	github.com/docker/docker v20.10.23+incompatible
```
Comment (vrothberg) [inline go.mod:25]: This will likely need to be pinned to the previous version to wait until imagebuilder is compatible with the new API

Comment (mtrmac) [reply go.mod:25]: This PR already includes an updated `openshift/imagebuilder`. (At least it seems to build and pass tests; it’s possible that it’s broken in some other known way I am unaware of.)

Comment (vrothberg) [reply go.mod:25]: Oh nice, thanks!

**Review by vrothberg — APPROVED** [review]
LGTM
@edsantiago PTAL

**Comment by edsantiago** [comment]
I'm sorry... this is not something I can realistically review, given my ignorance of the underlying modules.

**Comment by vrothberg** [comment]
@rhatdan PTAL

**Pull request merged**